In [2]:
import pandas as pd
df = pd.read_csv("cleaned_data.csv", parse_dates=["Order Date", "Ship Date"])
df.describe()

,Row ID,Order Date,Ship Date,Postal Code,Sales
count,9800.000000,9800,9800,9800.000000,9800.000000
mean,4900.500000,2017-05-01 05:13:51.673469440,2017-05-05 04:17:52.653061120,55217.343265,230.769059
min,1.000000,2015-01-03 00:00:00,2015-01-07 00:00:00,1040.000000,0.444000
25%,2450.750000,2016-05-24 00:00:00,2016-05-27 18:00:00,23223.000000,17.248000
50%,4900.500000,2017-06-26 00:00:00,2017-06-29 00:00:00,57551.000000,54.490000
75%,7350.250000,2018-05-15 00:00:00,2018-05-19 00:00:00,90008.000000,210.605000
max,9800.000000,2018-12-30 00:00:00,2019-01-05 00:00:00,99301.000000,22638.480000
std,2829.160653,NaN,NaN,32066.750532,626.651875


In [3]:
# Outliers in Sales (IQR method)
q1, q3 = df["Sales"].quantile([0.25, 0.75])
iqr = q3 - q1
outliers = df[(df["Sales"] < q1 - 1.5*iqr) | (df["Sales"] > q3 + 1.5*iqr)]
print("Outliers:", len(outliers))
print(outliers.sort_values("Sales", ascending=False)[["Product Name", "Category", "Sales"]].head(5))

Outliers: 1145
                                           Product Name    Category      Sales
2697  Cisco TelePresence System EX90 Videoconferenci...  Technology  22638.480
6826              Canon imageCLASS 2200 Advanced Copier  Technology  17499.950
8153              Canon imageCLASS 2200 Advanced Copier  Technology  13999.960
2623              Canon imageCLASS 2200 Advanced Copier  Technology  11199.968
4190              Canon imageCLASS 2200 Advanced Copier  Technology  10499.970


In [4]:
df["Month"] = df["Order Date"].dt.month_name()
df["Year"] = df["Order Date"].dt.year

print("Sales by Category:\n", df.groupby("Category")["Sales"].sum().sort_values(ascending=False), "\n")
print("Sales by Region:\n", df.groupby("Region")["Sales"].sum().sort_values(ascending=False), "\n")
print("Sales by Segment:\n", df.groupby("Segment")["Sales"].sum().sort_values(ascending=False), "\n")
print("Top 5 Sub-Categories:\n", df.groupby("Sub-Category")["Sales"].sum().sort_values(ascending=False).head(5), "\n")
print("Sales by Year:\n", df.groupby("Year")["Sales"].sum(), "\n")
print("Best Months:\n", df.groupby("Month")["Sales"].sum().sort_values(ascending=False).head(3))

Sales by Category:
 Category
Technology         827455.8730
Furniture          728658.5757
Office Supplies    705422.3340
Name: Sales, dtype: float64 

Sales by Region:
 Region
West       710219.6845
East       669518.7260
Central    492646.9132
South      389151.4590
Name: Sales, dtype: float64 

Sales by Segment:
 Segment
Consumer       1.148061e+06
Corporate      6.884941e+05
Home Office    4.249822e+05
Name: Sales, dtype: float64 

Top 5 Sub-Categories:
 Sub-Category
Phones     327782.448
Chairs     322822.731
Storage    219343.392
Tables     202810.628
Binders    200028.785
Name: Sales, dtype: float64 

Sales by Year:
 Year
2015    479856.2081
2016    459436.0054
2017    600192.5500
2018    722052.0192
Name: Sales, dtype: float64 

Best Months:
 Month
November     350161.7110
December     321480.1695
September    300103.4117
Name: Sales, dtype: float64


In [5]:
print(df[["Sales", "Postal Code"]].corr())

                Sales  Postal Code
Sales        1.000000    -0.025588
Postal Code -0.025588     1.000000


Task 2: Exploratory Data Analysis

Key Findings
- Technology is the top category by sales (about 827K), followed by Furniture and Office Supplies.
- West is the strongest region (about 710K); South is the weakest (about 389K).
- Consumer is the most valuable segment (about 1.15M), ahead of Corporate and Home Office.
- Phones, Chairs and Storage are the top-selling sub-categories.
- Sales dipped slightly in 2016, then grew strongly in 2017 and 2018.
- November, December and September are the best months, showing a year-end seasonal peak.
- 1,145 outliers were found using the IQR method. They are high-value Technology orders (e.g., Cisco TelePresence, Canon copiers), so they are genuine sales and were kept.
- Sales has almost no correlation with Postal Code (-0.03).

Conclusion: Sales are driven by Technology and the Consumer segment, peak at year-end, and the South region has the most room for growth.
